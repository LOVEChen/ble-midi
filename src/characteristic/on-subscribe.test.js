/* eslint-env jest */
jest.mock('../util/logger');

test('should onSubscribe(maxValueSize, updateValueCallback)', () => {
  const mockLog = require('../util/logger').default;
  const onSubscribe = require('./on-subscribe').default;
  const maxValueSize = jest.fn();
  const updateValueCallback = jest.fn();
  onSubscribe(maxValueSize, updateValueCallback);
  expect(mockLog.mock).toMatchSnapshot();
  expect(mockLog.mock.calls.length).toEqual(1);
  expect(mockLog.mock.calls[0].length).toEqual(3);
  expect(mockLog.mock.calls[0][1]).toEqual(maxValueSize);
  expect(mockLog.mock.calls[0][2]).toEqual(updateValueCallback);
});
