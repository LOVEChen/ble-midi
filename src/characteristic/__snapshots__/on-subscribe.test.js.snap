// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should onSubscribe(maxValueSize, updateValueCallback) 1`] = `
Object {
  "calls": Array [
    Array [
      "onSubscribe",
      [Function],
      [Function],
    ],
  ],
  "instances": Array [
    undefined,
  ],
}
`;
